luigi/worker.py

```diff
@@ -238,8 +238,11 @@ class DequeQueue(collections.deque):
     deque wrapper implementing the Queue interface.
     """
 
-    put = collections.deque.append
-    get = collections.deque.pop
+    def put(self, *args, **kwargs):
```

Comment:
It only needs to handle timeouts if it's using multiple processes. The worker will periodically check for new work while waiting for jobs to finish. If you're running with only one process, the timeout doesn't mean anything. Get and put will always return immediately. It's correct to throw out the timeout argument, but it would also be reasonable not to indiscriminately ignore any keyword argument and just match the signatures of the multiprocessing.Queue functions:

    put(obj[, block[, timeout]])
    get([block[, timeout]])

Just adding optional block and timeout arguments that you ignore looks more intentional and makes it more clear that you're not throwing out important arguments. If someone were to call DequeQueue().put(obj=x), your code would throw out the obj argument and result in an error.